### PR TITLE
[APB-3538][JR] FSM client journey: correcting back behaviour from /check-answers to /consent/individual

### PR DESCRIPTION
**Pull request by johnrevill**

### CODE REVIEW

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/ClientInvitationJourneyController.scala [hunk 1]
```diff
@@ -96,7 +96,7 @@ class ClientInvitationJourneyController @Inject()(
   }
 
   def showConsentIndividual = actionShowStateWhenAuthorised(AsClient) {
-    case _ =>
+    case _: SingleConsent =>
```
Comment (mariannepearson0) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/ClientInvitationJourneyController.scala]: by doing this are you saying you can be in any state and go to the single consent page. The backlink for breadcrumbs method should revert the state and show the page, is this not working?

Comment (johnrevill) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/ClientInvitationJourneyController.scala]: I'm not that familiar with this, but I think I meant to say that if you arrive at this action, then pick  out the last `SingleConsent` state in the breadcrumb history and render that state.

The backlink for breadcrumbs method was working when on the confirm-answers page and was giving the correct URL for the previous `/consent/individual` page.

However, following that link then makes a request that gets through to this `showConsentIndividual` action whilst the `Confirm` state is still at the head of the breadcrumbs list. 
So when the state filter on line 99 was allowing all states with `case _ =>`, it was coming to the `Confirm` state first and choosing that as the thing to display. So then it just displayed the `/confirm-answers` page again.

Should there be other states in that filter do you think?

Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/ClientInvitationJourneyController.scala]: but you don't want to be able to show the single consent page if you are not in the SingleConsent state. Ummmm so the url was right but the state wasn't being reverted?

Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/ClientInvitationJourneyController.scala]: oh sorry i completely confused myself here and thought you were removing the type

**Pull request merged**